src/ngraph/node_output.cpp

```diff
@@ -218,17 +233,53 @@ namespace ngraph
         }
     }
 
+    NodeVector Output<const Node>::get_users(bool check_is_used) const
+    {
+        NodeVector result;
+        for (auto input : get_target_inputs())
+        {
+            Node* input_node = input.get_node();
+            if (!check_is_used || is_used(input_node))
+            {
+                result.push_back(input_node->shared_from_this());
+            }
+        }
+        return result;
+    }
+
     std::ostream& operator<<(std::ostream& out, const Output<Node>& output)
     {
-        return output.get_node()->write_description(out, 0)
-               << "[" << output.get_index() << "]:" << output.get_element_type()
-               << output.get_partial_shape();
+        if (is_type<op::GetOutputElement>(output.get_node()))
+        {
+            auto captured_output = output.get_node()->input(0).get_source_output();
```

Comment:
Use a virtual function on `Node` -- then all the GOE stuff disappears when GOE goes away, and other nodes can provide enhanced printing (I think there are one or two that already do). This would also reduce duplication in the const node case below.

Reply:
Rewritten